Cargo.toml

```diff
@@ -40,6 +40,7 @@ sqlparser_derive = { version = "0.2.0", path = "derive", optional = true }
 simple_logger = "5.0"
 matches = "0.1"
 pretty_assertions = "1"
+itertools = "0.13"
```

Comment:
I'm not a huge fan of adding a new dependency...

Reply:
In the meantime, I will rewrite it without the use of itertools. Do let me know which variant you prefer.

Reply:
I think your version is okay. I would have written nested for loops, though.

Reply:
Ok! I find the flattened for-loops a bit easier to read, as otherwise, it gets to a very high level of indentation. Do let me know whether you have any more notes on the PR.